addons/mail/static/src/models/attachment_viewer/attachment_viewer.js

```diff
@@ -25,6 +25,24 @@ function factory(dependencies) {
         // Private
         //----------------------------------------------------------------------
 
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeAttachmentUrl() {
+            if (!this.attachment) {
+                return;
+            }
+            if (!this.attachment.accessToken && this.attachment.originThread && this.attachment.originThread.model === 'mail.channel') {
+                if (this.attachment.isPdf) {
+                    return `/web/static/lib/pdfjs/web/viewer.html?file=/mail/channel/${this.attachment.originThread.id}/attachment/${this.attachment.id}`;
+                }
+                return `/mail/channel/${this.attachment.originThread.id}/attachment/${this.attachment.id}`;
+            }
+            const accessToken = this.attachment.accessToken ? `?access_token=${this.attachment.accessToken}` : '';
+            return `/web/content/${this.attachment.id}${accessToken}`;
+        }
```

Comment:
Thank you for the update.

This does not necessarily need to be in `attachment_view`, as it doesn't use any field of the view if I'm seeing it correctly.

It could be directly in attachment model itself, actually this should be the result of `_computeDefaultSource`, adapted for the case when it is a channel.